src/wallet/rpcwallet.cpp

```diff
@@ -2389,33 +2389,72 @@ UniValue lockunspent(const JSONRPCRequest& request)
 
     RPCTypeCheckArgument(request.params[1], UniValue::VARR);
 
-    UniValue outputs = request.params[1].get_array();
-    for (unsigned int idx = 0; idx < outputs.size(); idx++) {
-        const UniValue& output = outputs[idx];
-        if (!output.isObject())
+    const UniValue output_params = request.params[1];
+
+    // Create and validate the COutPoints first.
+
+    std::vector<COutPoint> outputs;
+    outputs.reserve(output_params.size());
+
+    for (unsigned int idx = 0; idx < output_params.size(); idx++) {
+         const UniValue& param = output_params[idx];
+
+        if (!param.isObject()) {
```

Comment:
You can remove this check entirely. Type checking is done by the `UniValue.get_xxx()` functions, and returns almost an identical error to that given by this explicit check.

With the explicit check:

```
→ bitcoin-cli -rpcwallet=w1 lockunspent true [\"string\"]
error code: -8
error message:
Invalid parameter, expected
```

without:

```
→ bitcoin-cli -rpcwallet=w1 lockunspent true [\"string\"]
error code: -1
error message:
JSON value is not an object as expected
```
The implicit `get_xxx()` check is used extensively in other RPCs.

You can also combine the line below with above:

```
const UniValue& o = request.params[1].get_obj();
```